### ISSUE TITLE
Make every possibly long running task cancellable, with a progress indicator

### ISSUE THREAD

**Issue by jpmonettas** [flow-storm/flow-storm-debugger#143]
Currently just the search functionality is cancellable. It would be nice to also include things like function calls, printers, power-stepping etc. Every time one of this tasks is running a progress indicator or at least a spinner should show.

Making this possible would require some infrastructure first, to make it easier to support this on Clojure and ClojureScript.

One possible solution is to create a "cancellable entry processing batching system". I think if functionality can be expressed via a reduction step over the timeline entries (a transducer), then a system could run this reduction step in batches checking between them if interruption is needed. This can be implemented in Clojure using .isInterruped and in ClojureScript via a recursion with (js/setTimeout ... 0).

**Comment by maxrothman** [comment]
Once this is done, de/selecting argument checkboxes for the arguments pane of the function list view could refresh the view without requiring the user to double-click the fn list item again.

**Comment by jpmonettas** [comment]
@maxrothman I'll create another issue for the "de/selecting argument checkboxes for the arguments pane of the function list view could refresh the view without requiring the user to double-click the fn list item again" part.

**Comment by jpmonettas** [comment]
Created https://github.com/flow-storm/flow-storm-debugger/issues/158

**Comment by maxrothman** [comment]
Welp I should've read this before posting my comment on #147. Rather than messing around with buttons, would it be feasible to just have the detail view on the function list follow the list selection? My understanding was that the list required double-clicking because showing the detail view was slow and uncancelable. Now that it's cancelable, it wouldn't be so bad for a user to flip through a bunch of list items and have the detail view just follow their selection, right?

**Comment by jpmonettas** [comment]
@maxrothman We need to try this and play with task collection batches.  To improve perf we are allowing interruption on batch edges, so even if tasks are interruptible, it could take a little bit to interrupt, which will render keyboard navigation unusable.
But I guess the only way is to try it, and see how it feels.
